packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -2835,7 +2889,29 @@ class TextSelectionGestureDetector extends StatefulWidget {
 }
 
 class _TextSelectionGestureDetectorState extends State<TextSelectionGestureDetector> {
-  static int? _getDefaultMaxConsecutiveTap() => 2;
+  static int? _getDefaultMaxConsecutiveTap() {
+    switch(defaultTargetPlatform) {
+      // From observation, these platform's reset their tap count to 0 when the number of consecutive taps
+      // exceeds 3. For example on Debian Linux with GTK, when going past a triple click, on the fourth click the selection
```

Comment:
https://github.com/GNOME/gtk/blob/cefb0a3eb8f9a52283ce4368ccc78425db0914b0/gtk/gtktextview.h#L81-L94

GTK4 (and possibly previous versions too) selects the line not the paragraph on triple click.

Reply:
Thanks for catching that, I went back and double-checked and this matches the behavior I see.

Reply:
This PR matches this behavior now.